### PR TITLE
【NOMERGE】Profile load graph time

### PR DESCRIPTION
**Pull request by doombeaker**

### CODE REVIEW

File: examples/text_to_image_sdxl.py [hunk 1]
```diff
@@ -48,76 +47,72 @@
...
 image = base(
     prompt=args.prompt,
     height=args.height,
     width=args.width,
     num_inference_steps=args.n_steps,
     output_type=OUTPUT_TYPE,
 ).images
+flow._oneflow_internal.eager.Sync()
+end_time = time.time()
+print(f"{end_time-start_time}s elapsed: 1st infer")
```
Comment (doombeaker) [inline examples/text_to_image_sdxl.py:94]: 运行：

```bash
python examples/text_to_image_sdxl.py --compile_unet true --base /share_nfs/hf_models/stable-diffusion-xl-base-1.0
```

得到的结果比较奇怪：

虽然有 `load_graph` 的耗时（10 s 左右），但是第一次推理出图需要的时间，仍然需要 50s 以上。

```
Loading pipeline components...: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:01<00:00,  4.43it/s]
Compiling unet with oneflow.
1.2790741920471191s elapsed: compile
Loading from graph
    2.3534035682678223s elpased: assign of_module time
    0.03800535202026367s elpased: get_oneflow_graph time
  2.3914899826049805s elpased: dpl_graph time
  7.607182264328003s elpased: load_graph time
9.998764038085938s elapsed: unet.load_graph
Warmup with running graphs...
  0%|                                                                                                                                                                                      | 0/30 [00:00<?, ?it/s]    8.821487426757812e-06s elpased: assign of_module time
    0.03774738311767578s elpased: get_oneflow_graph time
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:53<00:00,  1.78s/it]
54.89510440826416s elapsed: 1st infer
Normal SDXL run...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:04<00:00,  7.40it/s]
4.648606061935425s elapsed: 2nd infer
```

Comment (strint) [reply examples/text_to_image_sdxl.py:94]: > 54.89510440826416s elapsed: 1st infer

怎么感觉在编译。

可以开下 ONEDIFF_BEBUG=1 来看是否触发了编译

Comment (strint) [reply examples/text_to_image_sdxl.py:94]: @doombeaker

**Comment by doombeaker** [comment]
经确认，不管是本 PR 的新增（简单）脚本，还是 https://github.com/siliconflow/onediff/blob/main/examples/text_to_image_sdxl_save_load.py 脚本。

不管有否 `load_graph`，第一次调用都会进入这个 `self._compile` 调用中，这个调用中会编译整个 graph

https://github.com/siliconflow/oneflow/blob/876c6d6b258e5ada564a0a71f490918317d764ab/python/oneflow/nn/graph/graph.py#L294-L295


这里有几个问题需要讨论和定位：

1. 之前有 **真正做到过** load_graph 后，调用时直接使用加载的 graph，整个端到端构图省掉编译时间的例子吗。如果有，那说明中间某些 commit，把功能改坏过
2. 一个 graph 调用时，到底是不是进行编译，判断条件是？ "graph 层就一个判断逻辑 if not self._is_compiled"，oneflow_compile 中还有些包装后的判断（如 `dpl_graph._is_raw_deployable_module`），感觉没有一致导致了这个问题。

**Comment by isidentical** [comment]
this is also something we observed quite recently, the first inference after loading the graph takes 40-50 seconds but I wasn't sure what happened. I think this is a new regression, but can't really point into the particular commit.

**Comment by doombeaker** [comment]
> this is also something we observed quite recently, the first inference after loading the graph takes 40-50 seconds but I wasn't sure what happened. I think this is a new regression, but can't really point into the particular commit.

sorry for that. I have confirmed this branch raise the bug: https://github.com/siliconflow/onediff/blob/814053b58e1f881c958d6fa4b3166d18a70ce368/src/onediff/infer_compiler/utils/args_tree_util.py#L40-L46

we will fix it soon.

**Comment by strint** [comment]
fix with https://github.com/siliconflow/onediff/pull/622

**Pull request closed without merging**